src/core/basetypes/MCData.h

```diff
@@ -28,7 +30,13 @@ namespace mailcore {
         static Data * dataWithCapacity(int capacity);
         static Data * dataWithContentsOfFile(String * filename);
         static Data * dataWithBytes(const char * bytes, unsigned int length);
-        
+
+         /* Create instance of Data with contents of given bytes without extra copying.
+         Memory ownership transferred from client code to the Data instance.
+         Data destructor will call bytesDeallocator function for memory releasing.
+         */
+        static Data * dataWithBytesTakeOwnership(char * bytes, unsigned int length, BytesDeallocator bytesDeallocator);
```

Comment:
This method is used in some other places. For example, in 
Data * IMAPSession::fetchMessageAttachment() method.